### PR TITLE
[Merged by Bors] - chore: rename `Complex.ofReal'` to `Complex.ofReal`

### PR DESCRIPTION
**Pull request by YaelDillies**
That's already how it was called in lemma names. Rename the existing `Complex.ofReal` to `Complex.ofRealHom`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Complex/Basic.lean [hunk 1]
```diff
@@ -76,10 +76,10 @@ theorem range_im : range im = univ :=
 /-- The natural inclusion of the real numbers into the complex numbers.
 The name `Complex.ofReal` is reserved for the bundled homomorphism. -/
```
Comment (grunweg) [inline Mathlib/Data/Complex/Basic.lean:77]: Please update that one also :-)

**Review by grunweg — COMMENTED** [review]
One small nit.

File: Mathlib/Data/Complex/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Complex/Basic.lean:77]: I simply deleted that sentence. It was there to justify the bad naming convention, which is now gone ✨

**Comment by loefflerd** [comment]
I think this is definitely a good idea, and it is at high risk of rapid bitrotting; so I support merging it pronto.

maintainer merge

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**